service/history/api/recordworkflowtaskstarted/api.go

```diff
@@ -74,6 +73,10 @@ func Invoke(
 				//  - Speculative WFT is lost (ScheduleToStart timeout for speculative WFT will recreate it).
 				return nil, serviceerror.NewNotFound("Workflow task not found.")
 			}
+			if workflowTask.Stamp != mutableState.GetExecutionInfo().GetWorkflowTaskStamp() {
+				// This happens when the workflow task was rescheduled.
+				return nil, serviceerrors.NewObsoleteMatchingTask("Workflow task stamp mismatch")
```

Comment:
Only place outside of task processing internals where we invalidate a workflow task with an outdated stamp now.

My thinking here is that when deploying a new version, all in-flight tasks and running workflows will have the default stamp, ie zero, and match. Only when the tasks are re-scheduled would they not match.